packages/@aws-cdk/aws-kms/lib/key.ts

```diff
@@ -217,15 +290,28 @@ export class Key extends KeyBase {
    * @param keyArn the ARN of an existing KMS key.
    */
   public static fromKeyArn(scope: Construct, id: string, keyArn: string): IKey {
+    const keyResourceName = Stack.of(scope).parseArn(keyArn).resourceName;
+    if (!keyResourceName) {
+      throw new Error(`KMS key ARN must be in the format 'arn:aws:kms:<region>:<account>:key/<keyId>', got: '${keyArn}'`);
+    }
+
     class Import extends KeyBase {
-      public keyArn = keyArn;
-      protected policy?: iam.PolicyDocument | undefined = undefined;
+      public readonly keyArn = keyArn;
+      public readonly keyId: string;
```

Comment:
why not just assign `keyResourceName` here?

Reply:
Because `keyResourceName` might be `undefined`, but `keyId` cannot be.